src/js/common/messageHelper.js

```diff
@@ -12,19 +12,25 @@ var gpii   = fluid.registerNamespace("gpii");
 
 fluid.registerNamespace("gpii.handlebars.helper.messageHelper");
 
-gpii.handlebars.helper.messageHelper.getHelper = function () {
+gpii.handlebars.helper.messageHelper.getHelper = function (that, serverAware) {
     return function (messageKey, dataOrRootContext, rootContext) {
         if (arguments.length < 2) {
             fluid.fail("You must call the 'messageHelper' helper with at least a message key.");
         }
         else {
-            // Pick up the message bundles from the root context, which is always the last argument.
-            var messages = fluid.get(rootContext || dataOrRootContext, "data.root.messages");
-
+            var messages;
+            if (serverAware) {
```

Comment:
We should have another meeting with @amb26 during our overlap, this is getting a bit out of hand and it's largely my fault.  I started this "server aware" pattern long ago, and it seems wrong headed for the long term.  I think we'd be better off migrating to something like named global variables for message bundles, templates, et cetera, or at least using something standard like a resource loader.  We should also at least figure out how/if we can avoid mixing events and model changes as we do currently.